external/avro/src/test/scala/org/apache/spark/sql/avro/AvroSuite.scala

```diff
@@ -1403,7 +1403,8 @@ abstract class AvroSuite
           df.write.option("avroSchema", avroSchema).format("avro").save(s"$tempDir/save2")
         }
         assertExceptionMsg[IncompatibleSchemaException](e,
-          "Cannot find field 'FOO' (at position 0) in Avro schema at top-level record")
+          "Cannot find SQL field(s) in Avro top-level record: " +
```

Comment:
The original error message is clear about the Avro schema doesn't have the field, while the new one makes me confused about which schema has missed field, Avro or Catalyst.

Also, showing one error at a time is OK.

@xkrogen Shall we keep the error message as it is? This PR can be just moving the schema validation to the `AvroSchemaHelper`. I am sorry but it's my personal taste as a non-native speaker.

Reply:
Ok, makes sense. I also agree that trying to phrase it properly when reporting multiple errors was more challenging. I think it's a tradeoff between more information vs. understandability of that information, and I'm find with going for better understandability. Updated.